### ISSUE TITLE
Emacs 29: Obsolete use of (point)

### ISSUE THREAD

**Issue by chasecaleb** [skeeto/emacs-web-server#27]
Native compiling with Emacs 29 gives the following error if i `(require 'simple-httpd)`:
```
Eager macro-expansion failure: (error \"`point' is an obsolete generalized variable; use `goto-char' instead.
```

I'm willing to make a PR to fix this, but it looks like there hasn't been any activity in this repo in quite awhile. @skeeto are you interested in merging a fix if I do it?

**Comment by skeeto** [comment]
Yeah, this is an extremely inconsiderate change by the Emacs developers 
who are not considering all the pointless make-work they're creating for 
other people. I'm hoping they'll realize the errors of their ways before 
the release of Emacs 29 so that I don't have to make hundreds of changes 
like this, for absolutely no benefit to anyone, across various projects. 
So I'm waiting until this behavior occurs in a stable release, not just 
test releases.

If this change does make it into Emacs 29 then I'll accept a PR with the 
fixes.

**Comment by chasecaleb** [comment]
Sounds good.

In the meantime, I _highly_ suggest adding your feedback to the Emacs mailing list. They aren't going to realize the "errors of their ways" if no one points out the pain it's causing, since of course Emacs maintainers can't audit every library in the wild :smile:

**Comment by chasecaleb** [comment]
@skeeto Emacs 29 is out and this change is still there. Do you want a PR, or do you want to fix it? I'm not familiar with the code of this project but it seems like a straightforward mechanical change so I imagine I can do it without difficulty.

**Comment by skeeto** [comment]
Thanks for the reminder, Caleb. No more warnings from Emacs >=29.1.

**Comment by chasecaleb** [comment]
Nice, thanks so much!
